backend/src/Altinn.Authorizationadmin/Altinn.Authorizationadmin/Services/Implementation/PolicyAdministrationPoint.cs

```diff
@@ -195,7 +204,16 @@ private async Task<bool> WriteDelegationPolicyInternal(string policyPath, List<R
                 try
                 {
                     // Check for a current delegation change from postgresql
-                    DelegationChange currentChange = await _delegationRepository.GetCurrentDelegationChange($"{org}/{app}", offeredByPartyId, coveredByPartyId, coveredByUserId);
+                    DelegationChange currentChange = new DelegationChange();
+
+                    string altinnAppId = string.Empty;
+                    if (!string.IsNullOrWhiteSpace(org) || !string.IsNullOrWhiteSpace(app))
```

Comment:
This can be moved to the if(resourceMatchType == ResourceAttributeMatchType.AltinnApp) scope earlier

Reply:
So you mean moving also the code that uses the altinnAppId out of this scope? For the only code that actually need to be in the if (leaseId != null) scope is the one that depends on the bloblease